src/firefly/js/charts/dataTypes/FireflyGenericData.js

```diff
@@ -72,3 +75,142 @@ function fetchData(chartId, traceNum, tablesource) {
     );
 }
 
+
+function addOtherChanges({changes, chartId, traceNum, tablesource, tableModel}) {
+    const type = get(getChartData(chartId), `data.${traceNum}.type`, 'scatter');
+
+    if (isScatter2d(type)) {
+        addScatterChanges({changes, chartId, traceNum, tablesource, tableModel});
+    }
+}
+
+/**
+ * Firefly added scatter changes
+ * @param {Object} p - parameters
+ * @param p.changes existing changes
+ * @param p.chartId chart id
+ * @param p.traceNum
+ * @param p.tablesource
+ * @param p.tableModel
+ */
+function addScatterChanges({changes, chartId, traceNum, tablesource, tableModel}) {
+
```

Comment:
Are we applying the changes only when it's not defined, or always?

Reply:
The enhancements are applied to scatter-like plots only. We apply firefly tooltips when hoverinfo is either not defined or is equal to 'text'. We apply labels to axes when they are not defined and there is only one trace.